pkg/registry/batch/rest/storage_batch.go

```diff
@@ -52,15 +54,21 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 }
 
 func (p RESTStorageProvider) v1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
-	version := batchapiv1.SchemeGroupVersion
-
 	storage := map[string]rest.Storage{}
+	p.AddV1JobsStorage(apiResourceConfigSource, restOptionsGetter, storage)
+	return storage
+}
+
+// Adds rest storage interfaces for v1/jobs resource to the given storage map.
+func (p RESTStorageProvider) AddV1JobsStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter, storage map[string]rest.Storage) {
```

Comment:
I want a method that also has the check. Will renaming this to AddV1JobsStorageIfEnabled help?